app/src/main/java/com/lumisdinos/gameoffifteen/domain/usecases/DeleteAllGamesUseCase.kt

```diff
@@ -0,0 +1,12 @@
+package com.lumisdinos.gameoffifteen.domain.usecases
+
+import com.lumisdinos.gameoffifteen.domain.repos.GameRepository
+import javax.inject.Inject
+
+class DeleteAllGamesUseCase @Inject constructor(
+    private val repository: GameRepository
+) {
+    suspend fun executeUseCase() {
+        return repository.deleteAllGames()
```

Comment:
you should be using subscribing asynchronous for all use cases (reactivex for example)

Reply:
I use inside this construction
CoroutineScope(Dispatchers.Main).launch {
            withContext(Dispatchers.IO) {
...
with other asynchronous calls since useCases never goes alone